security/pkg/k8s/controller/workloadsecret.go

```diff
@@ -294,7 +294,11 @@ func (sc *SecretController) citadelManagedObject(obj metav1.Object) bool {
 
 // Handles the event where a service account is added.
 func (sc *SecretController) saAdded(obj interface{}) {
-	acct := obj.(*v1.ServiceAccount)
+	acct, ok := obj.(*v1.ServiceAccount)
+	if !ok {
+		k8sControllerLog.Warnf("Failed to convert to serviceaccount object: %v", obj)
```

Comment:
under what condition the conversion will fail?

Reply:
please see the documentation for possible returned values:
https://github.com/kubernetes/client-go/blob/master/tools/cache/controller.go#L166-L184
> OnDelete will get the final state of the item if it is known, otherwise it will get an object of type DeletedFinalStateUnknown.

Reply:
`saAdded` is passed as `AddFunc`,  why OnDelete behavior matters here?

Reply:
https://sourcegraph.com/github.com/istio/istio@565cd1adeaa12e97b02af20f8c26d8bd14af550a/-/blob/security/pkg/k8s/controller/workloadsecret.go#L225